api/user.go

```diff
@@ -182,6 +182,62 @@ func (a *UserAPI) CreateUser(ctx *gin.Context) {
 	}
 }
 
+// Register creates a user who can't be admin
+// swagger:operation POST /registration user register
+//
+// Create a user.
+//
+// ---
+// consumes: [application/json]
+// produces: [application/json]
+// security: [clientTokenHeader: [], clientTokenQuery: [], basicAuth: []]
+// parameters:
+// - name: body
+//   in: body
+//   description: the user to add
+//   required: true
+//   schema:
+//     $ref: "#/definitions/UserWithPass"
+// responses:
+//   200:
+//     description: Ok
+//     schema:
+//         $ref: "#/definitions/User"
+//   400:
+//     description: Bad Request
+//     schema:
+//         $ref: "#/definitions/Error"
+//   401:
+//     description: Unauthorized
+//     schema:
+//         $ref: "#/definitions/Error"
+//   403:
+//     description: Forbidden
+//     schema:
+//         $ref: "#/definitions/Error"
+func (a *UserAPI) Register(ctx *gin.Context) {
```

Comment:
Please combine this method with the existing create user api, as the signature is exactly the same. 

The admin property on the user can then only be set if the current user is an administrator.

Please also add tests for the newly added functionality.

Reply:
I've been helping @p1gp1g on this PR, and I'm wondering, is this approach kind of what you mean here? [1]
The code isn't fully working, but I wanted to check in with you before testing/cleaning/debugging etc.
[1] https://github.com/karmanyaahm/server/commit/c0b4a897723114d2a4a12d84000d2138d161ceef

Reply:
Yes, this goes in the right direction.